scantool/diag_l0_elm.c

```diff
@@ -743,13 +752,77 @@ elm_initbus(struct diag_l0_device *dl0d, struct diag_l1_initbus_args *in)
 					break;
 			}
 
+			//set init address
+			if ((dev->elmflags & ELM_323_BASIC) && (in->addr != 0x33)) {
+				fprintf(stderr, FLFMT "elm_initbus: ELM323 doesn't support target address %02X", FL, in->addr);
+				return diag_iseterr(DIAG_ERR_GENERAL);
+			} else {
+				uint8_t iia[15];
+				sprintf((char *) iia, "ATIIA %02X\x0D", in->addr);
```

Comment:
I believe that most "ELM v1.5" clones support ATIIA. Some clones don't support ATIIA. Depending on the clone, they either respond with "?" or treat ATIIA as ATI.

The ELM's default target address for ISO9141 is 0x33, which is used for J1979 over ISO9141. The new ATIIA code treats a failure as soft if the address is 0x33, so it should still work for J1979 on clones that don't support ATIIA. However, I don't have one of those clones to test with.

KWP6227 isn't possible on clones without ATIIA.